src/gui/qgsfieldmappingmodel.cpp

```diff
@@ -94,64 +94,75 @@ QVariant QgsFieldMappingModel::data( const QModelIndex &index, int role ) const
 
-    if ( role == Qt::DisplayRole || role == Qt::EditRole )
+    switch ( role )
     {
-      switch ( static_cast<ColumnDataIndex>( col ) )
+      case Qt::DisplayRole:
+      case Qt::EditRole:
       {
-        case ColumnDataIndex::SourceExpression:
-        {
-          return f.expression;
-        }
-        case ColumnDataIndex::DestinationName:
-        {
-          return f.field.displayName();
-        }
-        case ColumnDataIndex::DestinationType:
-        {
-          return static_cast<int>( f.field.type() );
-        }
-        case ColumnDataIndex::DestinationLength:
-        {
-          return f.field.length();
-        }
-        case ColumnDataIndex::DestinationPrecision:
+        switch ( col )
         {
-          return f.field.precision();
+          case ColumnDataIndex::SourceExpression:
+          {
+            return f.expression;
+          }
+          case ColumnDataIndex::DestinationName:
+          {
+            return f.field.displayName();
+          }
+          case ColumnDataIndex::DestinationType:
+          {
+            return static_cast<int>( f.field.type() );
+          }
+          case ColumnDataIndex::DestinationLength:
+          {
+            return f.field.length();
+          }
+          case ColumnDataIndex::DestinationPrecision:
+          {
+            return f.field.precision();
+          }
+          case ColumnDataIndex::DestinationConstraints:
+          {
+            return constraints != 0 ? tr( "Constraints active" ) : QString();
+          }
         }
-        case ColumnDataIndex::DestinationConstraints:
-        {
-          return constraints != 0 ? tr( "Constraints active" ) : QString();
-        }
-      }
-    }
-    else if ( role == Qt::ToolTipRole &&
-              col == static_cast<int>( ColumnDataIndex::DestinationConstraints ) &&
-              constraints != 0 )
-    {
-      QStringList constraintDescription;
-      if ( constraints.testFlag( QgsFieldConstraints::Constraint::ConstraintUnique ) )
-      {
-        constraintDescription.push_back( tr( "Unique" ) );
+        break;
       }
-      if ( constraints.testFlag( QgsFieldConstraints::Constraint::ConstraintNotNull ) )
+      case Qt::ToolTipRole:
       {
-        constraintDescription.push_back( tr( "Not null" ) );
+        if ( col == ColumnDataIndex::DestinationConstraints &&
+             constraints != 0 )
+        {
+          QStringList constraintDescription;
+          if ( constraints.testFlag( QgsFieldConstraints::Constraint::ConstraintUnique ) )
+          {
+            constraintDescription.push_back( tr( "Unique" ) );
+          }
+          if ( constraints.testFlag( QgsFieldConstraints::Constraint::ConstraintNotNull ) )
+          {
+            constraintDescription.push_back( tr( "Not null" ) );
+          }
+          if ( constraints.testFlag( QgsFieldConstraints::Constraint::ConstraintNotNull ) )
```

Comment:
`ConstraintExpression`

Reply:
what?

Reply:
This if should be with `ConstraintExpression`, `ConstraintNotNull` is in the previous if.

Reply:
oh, I see, thank you.